src/transformState.js

```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        for (const add in key.extraData) {
+          state[add] = key.extraData[add];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const remove of key.keysToRemove) {
+          delete state[remove];
+        }
+        break;
+
+      case 'clear':
+        for (const property in state) {
+          delete state[property];
+        }
+        break;
+
+      default:
```

Comment:
If no case is matched then the default code is executed (if it exists)!
You have it in your code, but it does nothing.